### PR TITLE
Langfuse Evaluation Suite with Public Scores

### LINKED ISSUE #157: portfolio-agent: langfuse eval suite with public scores on /about
## Goal
Articulate "how I know this LLM works" — the most common gap on AI portfolios. **LangFuse** (self-hostable, vendor-portable) over LangSmith for portability.

## Eval datasets
- [ ] Golden Q&A (~30 items): publications, blog topics, project skills, bio facts, resume specifics
- [ ] Adversarial set (~15 items): off-topic, prompt-injection attempts, ambiguous queries → expected refusals
- [ ] Citation-accuracy set (~20 items): questions with verifiable source attribution
- [ ] Edge cases: stale info ("what's Sean working on now?"), opinion-eliciting, comparative claims

## Eval metrics
- [ ] Hallucination rate (LLM-as-judge with explicit grounding rubric, judge ≠ agent's LLM)
- [ ] Retrieval precision@5 against golden chunks
- [ ] Refusal correctness (precision/recall on adversarial)
- [ ] Citation validity (cited URL contains the cited claim — programmatic check)
- [ ] Latency p50/p95
- [ ] Cost per query

## Public display
- [ ] CI job runs evals nightly, writes score JSON to public location (GCS bucket or commit to repo)
- [ ] `/about` page renders latest scores with sparklines (last 30 days)
- [ ] Link to LangFuse trace samples for transparency (one good, one refusal example)

## Why this matters
"I can articulate how I evaluate my LLM" is what F500 ML platform teams hire for. Most candidates skip this.

### PR DESCRIPTION
**Pull request by seanbearden**
I have implemented the Langfuse evaluation suite as requested. This includes:
1.  **Evaluation Datasets**: 75+ test cases covering bio facts, resume specifics, publications, adversarial attempts, and edge cases.
2.  **Evaluation Runner**: A Python script (`evals/run_evals.py`) that executes the suite, uses GPT-4o as a judge for qualitative metrics, and tracks performance in Langfuse. It calculates p50 and p95 latencies and captures trace IDs.
3.  **Nightly CI**: A new GitHub Action that runs every night, merging the latest scores into a 30-day history file stored in GCS.
4.  **Frontend Integration**: The `/about` page now features an "LLM Evaluation Suite" section with real-time data fetching, SVG sparklines for historical trends, and links to representative Langfuse traces.
5.  **Infrastructure & Verification**: Updated Terraform for CORS support and verified all changes with Vitest and visual inspection.

Fixes #157

---
*PR created automatically by Jules for task [11877372075874050999](https://jules.google.com/task/11877372075874050999) started by @seanbearden*

### CODE REVIEW

File: evals/run_evals.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            "citation_validity": sum(r["score"] for r in results if r["dataset"] == "citations") / len([r for r in results if r["dataset"] == "citations"]),
+            "latency_p50": float(np.percentile(all_latencies, 50)),
+            "latency_p95": float(np.percentile(all_latencies, 95)),
+            "avg_cost": sum(r["cost"] for r in results) / len(results)
+        },
+        "samples": {
+            "good": f"https://cloud.langfuse.com/project/{os.getenv('LANGFUSE_PROJECT_ID', 'clog')}/traces/{good_sample.get('trace_id', '')}",
+            "refusal": f"https://cloud.langfuse.com/project/{os.getenv('LANGFUSE_PROJECT_ID', 'clog')}/traces/{refusal_sample.get('trace_id', '')}"
+        }
+    }
```
[the remark this answers is not in the document]
Comment (seanbearden) [reply evals/run_evals.py]: Applied in be6198999bee55d5f7e8ff9b9fb5e9ebc8ba7d41 — almost exactly your suggestion. Added the `avg()` helper, early-return on empty `results`, and split out the `good_id` / `refusal_id` lookups so the format string isn't doing the work.

File: frontend/src/components/EvalSuite.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  const metrics: { key: keyof EvalMetrics; label: string; format: (v: number) => string; inverse?: boolean }[] = [
+    { key: "hallucination_rate", label: "Hallucination Rate", format: (v) => `${(v * 100).toFixed(1)}%`, inverse: true },
+    { key: "retrieval_precision", label: "Retrieval Precision@5", format: (v) => `${(v * 100).toFixed(1)}%` },
+    { key: "refusal_correctness", label: "Refusal Correctness", format: (v) => `${(v * 100).toFixed(1)}%` },
+    { key: "citation_validity", label: "Citation Validity", format: (v) => `${(v * 100).toFixed(1)}%` },
+    { key: "latency_p50", label: "Latency (p50)", format: (v) => `${v.toFixed(2)}s`, inverse: true },
+    { key: "latency_p95", label: "Latency (p95)", format: (v) => `${v.toFixed(2)}s`, inverse: true },
+    { key: "avg_cost", label: "Cost / Query", format: (v) => `$${v.toFixed(4)}`, inverse: true },
+  ];
```
[the remark this answers is not in the document]
Comment (seanbearden) [reply frontend/src/components/EvalSuite.tsx]: Hoisted in be6198999bee55d5f7e8ff9b9fb5e9ebc8ba7d41 — `METRICS` is now at module scope and renamed in upper case to flag it as a constant.
